### PR TITLE
feat(company-list-ui): implement CompanyListPage with search, sort functionality, and company item UI display

### PR DESCRIPTION
**Pull request by iraboldyrieva**
## Describe your changes

- Implemented CompanyListPage with features:
-Search functionality to filter companies by name.
-Sort button to toggle sorting order (ascending/descending).
-Display of each company item with an avatar, name, email, and action icons (edit and navigate).

## Issue ticket code (and/or) and link

- [SCRUM-7: Create UI Company list](https://codecraftersintership.atlassian.net/browse/SCRUM-7)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [x] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

### CODE REVIEW

File: src/pages/CompanyListPage/index.tsx [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+  const { t } = useTranslation();
+  const [searchTerm, setSearchTerm] = useState<string>('');
+
+  const [sortOrder, setSortOrder] = useState<'asc' | 'desc'>('asc');
+
+  const companies: Company[] = [
+    { id: 1, name: 'Company Alpha', email: 'alpha@company.com' },
+    { id: 2, name: 'Company Beta', email: 'beta@company.com' },
+    { id: 3, name: 'Company Gamma', email: 'gamma@company.com' },
+  ];
```
Comment (lyudad) [inline src/pages/CompanyListPage/index.tsx]: move above the component, we dont need reinitialise with each rerender

File: src/pages/CompanyListPage/index.tsx [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+      const nameA = a.name.toLowerCase();
+      const nameB = b.name.toLowerCase();
+
+      if (nameA < nameB) return sortOrder === 'asc' ? -1 : 1;
+      if (nameA > nameB) return sortOrder === 'asc' ? 1 : -1;
+      return 0;
+    });
+
+    return sorted;
+  }, [companies, searchTerm, sortOrder]);
```
Comment (lyudad) [inline src/pages/CompanyListPage/index.tsx]: this should be done on be, as there is pagination, you just need to make right request

Comment (iraboldyrieva) [reply src/pages/CompanyListPage/index.tsx]: will do.

File: src/pages/CompanyListPage/index.tsx (hunk 1 above)
Comment (lyudad) [reply src/pages/CompanyListPage/index.tsx]: nothing changed

File: src/pages/CompanyListPage/index.tsx (hunk 2 above)
Comment (lyudad) [reply src/pages/CompanyListPage/index.tsx]: it shill there

Comment (iraboldyrieva) [reply src/pages/CompanyListPage/index.tsx]: sorry, my bad. i'll be more attentive.

File: src/pages/CompanyListPage/index.tsx [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+          companies.map((company) => (
+            <CompanyItem
+              key={company.id}
+              company={company}
+              onEdit={() => console.log(`Edit: ${company.id}`)}
+              onNavigate={() => console.log(`More: ${company.id}`)}
+            />
+          ))
+        ) : (
+          <Typography>No companies yet</Typography>
```
Comment (lyudad) [inline src/pages/CompanyListPage/index.tsx]: 1.3.3

File: src/pages/CompanyListPage/usePaginationAndSorting.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import React, { useState } from 'react';
+
+const usePaginationAndSorting = (initialSortOrder: 'asc' | 'desc' = 'asc') => {
+  const [page, setPage] = useState(1);
+  const [sortOrder, setSortOrder] = useState(initialSortOrder);
```
Comment (lyudad) [inline src/pages/CompanyListPage/usePaginationAndSorting.tsx]: type

Comment (iraboldyrieva) [reply src/pages/CompanyListPage/usePaginationAndSorting.tsx]: done

File: src/pages/CompanyListPage/index.tsx (hunk 3 above)
Comment (iraboldyrieva) [reply src/pages/CompanyListPage/index.tsx]: done

**Pull request merged**